### PR TITLE
Add buffer time to perf test

### PR DESCRIPTION
**Pull request by OwenKephart**
## Summary & Motivation

As title. This is a consequence of getting rid of the special-purpose PartitionKeysTimeWindowPartitionsSubset. Long story short, these perf tests are not very representative of real-world perf scenarios, and mostly exist as a backstop in case anything massively changes. This change is also completely irrelevant to the Declarative Automation perf characteristics, which are more important and have their own separate perf tests.

TL;DR: We'll probably remove this test in the near future, for now we're just making it stop failing

## Changelog

NOCHANGELOG

### CODE REVIEW

**Comment by OwenKephart** [comment]
* **#25915** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/25915?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `master`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @OwenKephart and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

**Review by yuhan — APPROVED** [review]
